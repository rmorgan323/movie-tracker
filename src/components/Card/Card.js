import React, { Component } from 'react';
import './Card.css';

const Card = ( {image, title, releaseDate, voteAverage, overview} ) => {
	return (
		<div className="card">
			<div><img src={image} alt={title} /></div>
			<h2>{title}</h2>
			<h4>Released: {releaseDate}</h4>
			<h4>Average Review: {voteAverage}</h4>
			<h4 className="overview">Overview: {overview}</h4>
		</div>
	)
}

export default Card;