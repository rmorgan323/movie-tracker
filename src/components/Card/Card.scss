.card {
	width: 300px;
	background-color: lightgrey;
	padding: 10px;
	margin: 10px;

	img {
		width: 280px;
	}

	h2 {
		text-align: center;
	}

	h2, h4 {
		margin: 8px 0;
	}

	.overview {
		height: 91px;
		overflow: scroll;
	}
}