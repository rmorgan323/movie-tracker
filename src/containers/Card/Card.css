.card {
  width: 300px;
  background-color: rgba(60, 60, 60, 0);
  margin: 10px;
  height: 450px;
  cursor: pointer;
  position: relative;
  color: rgba(255, 255, 255, 0);
  transition: background-color .3s ease-in-out;
  transition: color .1s ease-in-out; }
  .card img {
    width: 300px;
    opacity: 1 !important;
    position: absolute;
    z-index: -1; }
  .card h2 {
    text-align: center;
    margin: 14px 12px 10px;
    font-weight: 300; }
  .card h4 {
    line-height: 1.3;
    margin: 8px 12px;
    font-family: 'Sanchez', serif; }
  .card .overview {
    height: 300px;
    overflow: scroll; }
  .card button {
    background: none;
    border: none;
    cursor: pointer;
    position: absolute;
    bottom: 5px;
    right: 5px;
    outline: none; }
  .card .star-outline {
    background-image: url("star-outline.svg");
    background-repeat: no-repeat;
    height: 70px;
    opacity: .7;
    width: 70px; }
  .card .star-outline:hover {
    background: none;
    background-image: url("star-outline-red.svg");
    background-repeat: no-repeat;
    opacity: 1; }
  .card .star-solid {
    background-image: url("star-solid.svg");
    background-repeat: no-repeat;
    height: 70px;
    width: 70px;
    opacity: 1; }

.card:hover {
  background-color: rgba(60, 60, 60, 0.88);
  color: rgba(255, 255, 255, 0.82);
  transition: background-color 1s ease-in-out;
  transition: color 1s ease-in-out; }

.card-favorited {
  background-color: red; }
