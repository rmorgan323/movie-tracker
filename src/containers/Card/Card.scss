.card {
	width: 300px;
	background-color: rgba(60,60,60, 0);
	margin: 10px;
	height: 450px;
	cursor: pointer;
	position: relative;
	color: rgba(255,255,255, 0);
	transition: background-color 4s ease-in-out;
	transition: color .1s ease-in-out;

	img {
		width: 300px;
		opacity: 1 !important;
		position: absolute;
		z-index: -1;
	}

	h2 {
		text-align: center;
		margin: 14px 12px 10px;
		font-weight: 300;
	}

	h4 {
		line-height: 1.3;
		margin: 8px 12px;
		font-family: 'Sanchez', serif;
	}

	.overview {
		height: 300px;
		overflow: scroll;
	}

	button {
		background: none;
		border: none;
		cursor: pointer;
		position: absolute;
		bottom: 5px;
		right: 5px;
		outline: none;
	}

	.star-outline {
		background-image: url('star-outline.svg');
		background-repeat: no-repeat;
		height: 70px;
		opacity: .7;
		width: 70px;
	}

	.star-outline:hover {
		background: none;
		background-image: url('star-outline-red.svg');
		background-repeat: no-repeat;
		opacity: 1;
	}

	.star-solid {
		background-image: url('star-solid.svg');
		background-repeat: no-repeat;
		height: 70px;
		width: 70px;
		opacity: 1;
	}
}

.card:hover {
	background-color: rgba(60,60,60, .88);
	color: rgba(255,255,255, .82);
	transition: background-color 1s ease-in-out;
	transition: color 1s ease-in-out;
}

::-webkit-scrollbar {
	background: transparent;
	width: 0px;
}


