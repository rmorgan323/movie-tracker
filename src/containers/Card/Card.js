import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../../actions/';
import PropTypes from 'prop-types';
import './Card.css';

export const Card = props => {
  const {
    movieId,
    image,
    userId,
    title,
    releaseDate,
    voteAverage,
    overview
  } = props;

  const movieBundle = {
    movieId,
    image,
    userId,
    title,
    releaseDate,
    voteAverage,
    overview
  };

  const favorite = props.user.userInfo.favorites.find(
    favorite => favorite.title === movieBundle.title
  )
    ? true
    : false;

  const handleFavorites = (props, movieBundle) => {
    if (
      props.user.userInfo.favorites.find(
        favorite => favorite.title === movieBundle.title
      )
    ) {
      props.deleteFavorite(movieBundle);
    } else {
      props.addFavorite(movieBundle);
    }
  };

  return (
    <div className="card">
      <div>
        <img src={image} alt={title} />
      </div>

      <h2>{title}</h2>
      <h4>Released: {releaseDate}</h4>
      <h4>Average Review: {voteAverage}</h4>
      <h4 className="overview">Overview: {overview}</h4>
      <button
        onClick={() =>
          userId
            ? handleFavorites(props, movieBundle)
            : alert('hey idiot, why not sign up??')
        }
        className={favorite ? 'star-solid' : 'star-outline'}
      />
    </div>
  );
};

export const mapStateToProps = store => {
  return {
    user: store.user
  };
};

export const mapDispatchToProps = dispatch => {
  return {
    deleteFavorite: movieBundle => {
      dispatch(actions.deletePost(movieBundle));
    },
    addFavorite: movieBundle => {
      dispatch(actions.addPost(movieBundle));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Card);

Card.propTypes = {
  movieId: PropTypes.string,
  userId: PropTypes.string,
  title: PropTypes.string,
  releaseDate: PropTypes.string,
  voteAverage: PropTypes.string,
  overview: PropTypes.string,
  image: PropTypes.string,
  location: PropTypes.string,
  movies: PropTypes.array,
  user: PropTypes.object,
  deleteFavorite: PropTypes.func,
  addFavorite: PropTypes.func
};
