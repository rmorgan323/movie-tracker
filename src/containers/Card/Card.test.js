import { Card, mapStateToProps, mapDispatchToProps } from './Card';
import React from 'react';
import { shallow } from 'enzyme';

describe('Card tests', () => {

});
