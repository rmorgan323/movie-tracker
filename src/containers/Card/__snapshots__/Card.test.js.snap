// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    user={
        Object {
            "signedIn": false,
            "userInfo": Object {
              "email": "",
              "favorites": Array [],
              "id": "",
              "name": "",
              "password": "",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <img
                    alt={undefined}
                    src={undefined}
          />
</div>,
        <h2 />,
        <h4>
          Released: 
</h4>,
        <h4>
          Average Review: 
</h4>,
        <h4
          className="overview"
>
          Overview: 
</h4>,
        <button
          className="star-outline"
          onClick={[Function]}
/>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt={undefined}
            src={undefined}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Released: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Released: ",
          undefined,
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Average Review: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Average Review: ",
          undefined,
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Overview: ",
            undefined,
          ],
          "className": "overview",
        },
        "ref": null,
        "rendered": Array [
          "Overview: ",
          undefined,
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "star-outline",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <img
                        alt={undefined}
                        src={undefined}
            />
</div>,
          <h2 />,
          <h4>
            Released: 
</h4>,
          <h4>
            Average Review: 
</h4>,
          <h4
            className="overview"
>
            Overview: 
</h4>,
          <button
            className="star-outline"
            onClick={[Function]}
/>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt={undefined}
              src={undefined}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": undefined,
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Released: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Released: ",
            undefined,
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Average Review: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Average Review: ",
            undefined,
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Overview: ",
              undefined,
            ],
            "className": "overview",
          },
          "ref": null,
          "rendered": Array [
            "Overview: ",
            undefined,
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "star-outline",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
